sources/ghostdog/src/main.rs

```diff
@@ -319,6 +379,302 @@ lazy_static! {
+    let ib_device =
+        InfinibandDevice::from_str(ibstat_str).context(error::ParseIbstatOutputSnafu)?;
+    Ok(ib_device)
+}
+
+/// write_guid will read the two files that need to be coordinated, search them for
+/// existing references, and replace the reference if found. If the reference doesn't
+/// exist, it will append to the end of the file the correct reference. This uses a
+/// tempfile to avoid reading and writing to the same file and copies the temp file
+/// over the exiting file when complete.
+fn write_guid(guid: String) -> Result<()> {
+    let mut found_fabric_manager_port = false;
+    let mut found_nvlsm_port = false;
+
+    // Search and update fabricmanager.cfg
+    let mut fm_config = NamedTempFile::new().context(error::CreateTemporaryFileSnafu)?;
+
+    if let Ok(lines) = read_lines(FABRIC_MANAGER_CONFIG_PATH) {
+        for line in lines.map_while(std::result::Result::ok) {
+            if line.contains("FM_SM_MGMT_PORT_GUID") {
+                found_fabric_manager_port = true;
+                writeln!(fm_config, "FM_SM_MGMT_PORT_GUID={}", guid.clone()).context(
+                    error::WriteFileSnafu {
+                        path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+                    },
+                )?;
+            } else {
+                writeln!(fm_config, "{}", line).context(error::WriteFileSnafu {
+                    path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+                })?;
+            }
+        }
+    }
+    // If there was no reference replaced, append to the end of the file
+    if !found_fabric_manager_port {
+        writeln!(fm_config, "FM_SM_MGMT_PORT_GUID={}", guid.clone()).context(
+            error::WriteFileSnafu {
+                path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+            },
+        )?;
+    }
+
+    // Search and update nvlsm.conf
+    let mut nvlsm_config = NamedTempFile::new().context(error::CreateTemporaryFileSnafu)?;
+
+    if let Ok(lines) = read_lines(NVLSM_CONFIG_PATH) {
+        for line in lines.map_while(std::result::Result::ok) {
+            if line.contains("guid 0x") {
+                found_nvlsm_port = true;
+                writeln!(nvlsm_config, "guid {}", guid.clone()).context(error::WriteFileSnafu {
+                    path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+                })?;
+            } else {
+                writeln!(nvlsm_config, "{}", line).context(error::WriteFileSnafu {
+                    path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+                })?;
+            }
+        }
+    }
+    // If there was no reference replaced, append to the end of the file
+    if !found_nvlsm_port {
+        writeln!(nvlsm_config, "guid {}", guid.clone()).context(error::WriteFileSnafu {
+            path: FABRIC_MANAGER_CONFIG_PATH.to_string(),
+        })?;
+    }
+
+    // Wait until both writes have succeeded before copying to ensure a minimal chance of unmatched files
```

Comment:
I don't understand this comment - is something watching writes to these files via inotify? If so, isn't there a race even if it's "minimal"?